### ISSUE TITLE
A need to refactor the CellarRegistry

### ISSUE THREAD

**Issue by IceDragon200** [GrowthcraftCE/Growthcraft-1.7#88]
There is some voodoo going on in there, if you manage to turn on the linting features during compilation, your console will be assaulted by warnings.

Ranging from untyped Maps to mixed declarations (variables between method definitions)

This isn't all that urgent, but when we do get around to it (or when I get around it), it's gonna get a complete overhaul and face lift, splitting up the responsibilities it currently has.

So, what does the `CellarRegistry` do?
Well it registers your booze _AND_ it registers your plants for pressing _AND_ it registers your brewing recipes _AND_ for some reason it registers heat sources...

The fact we had to use _AND_ to describe this class is a call for concern, what happens when we need a new API?
Throw it in the `CellarRegistry`? If you answered yes, congrats now you added another _AND_, `yet another API`

tl;dr We split the `CellarRegistry` into smaller registries, it will remain as a host for the other registries.
Ultimately, the end result is its description should be "CellarRegistry houses references to other registrar systems used by Growthcraft". 
And each registrar's description will be as simple as "Does x", eg. "The BrewingRegistry manages brewing recipes."
We can debug each separately without hitting our heads with brewing and pressing when all you wanted was good old booze.

Lets put this on the TODO list, along with moving block variables into their own classes (they're littered on the Mod classes, which is another mess) as well as Config (which is also littered unto the Mod classes)

I can find problems but, I'm bad at making a milestone, so
@alatyami, @Sunconure11 WDYT?
